packages/cojson/src/coValues/group.ts

```diff
@@ -203,7 +262,75 @@ export class RawGroup<
       "trusting",
     );
 
+    console.log("Setting", `readKey`, "to", newReadKey.id, "in", this.id);
+
     this.set("readKey", newReadKey.id, "trusting");
+
+    for (const parent of this.getParentGroups()) {
```

Comment:
To give accounts read access to a group, we reveal the group's `readKey` to the account's (public, asymmetric) key.

To give parent groups read access to a child group, we now similarly reveal the child group's `readKey` to the parent group's `readKey`

And finally, when we rotate our `readKey` (because someone got kicked out), we also need to (recursively) rotate the `readKey`s of all child groups (so they are kicked out there as well)

Reply:
> To give parent groups read access to a child group, we now similarly reveal the child group's readKey to the parent group's (secret, symmetric) readKey

Wait, why a parent group should have read access to a child group?

Reply:
Because the roles should be inherited - anyone who is a reader in the parent should also be a reader in the child. By exposing the child key to the parent key we can efficiently share it with everyone who is a reader in the parent